From b7b981748d646e05d40f42d21d091239db5203fe Mon Sep 17 00:00:00 2001
From: GloriousEggroll <gloriouseggroll@gmail.com>
Date: Wed, 10 Jul 2024 14:56:32 -0600
Subject: [PATCH] don't force display and xauth envvars in user service, and
 pass existing envvars when running subprocess popen. Also properly split
 custom application command into array for popen.

---
 data/service/yumex-updater-systray.service | 2 --
 yumex/service/data.py                      | 8 ++++++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/data/service/yumex-updater-systray.service b/data/service/yumex-updater-systray.service
index 4d989fe..bcae073 100644
--- a/data/service/yumex-updater-systray.service
+++ b/data/service/yumex-updater-systray.service
@@ -8,8 +8,6 @@ Type=simple
 ExecStart=/usr/bin/yumex_updater_systray
 RestartSec=5
 Restart=on-failure
-Environment=DISPLAY=:0
-Environment=XAUTHORITY=%h/.Xauthority
 
 [Install]
 WantedBy=graphical-session.target
diff --git a/yumex/service/data.py b/yumex/service/data.py
index 91356f5..febfe05 100644
--- a/yumex/service/data.py
+++ b/yumex/service/data.py
@@ -24,6 +24,7 @@ import configparser
 import shutil
 import logging
 import subprocess
+import os
 
 from pathlib import Path
 from dataclasses import dataclass
@@ -43,7 +44,8 @@ logger = logging.getLogger("yumex_updater")
 def open_yumex(*args):
     """launch yumex"""
     logger.info(f"open_yumex {args}")
-    subprocess.Popen(["/usr/bin/yumex", "--update"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+    env = os.environ.copy()
+    subprocess.Popen(["/usr/bin/yumex", "--update"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, env=env)
 
 
 @dataclass
@@ -100,7 +102,9 @@ class Indicator:
 
     def on_clicked_custom(self, *args) -> None:
         """start custom updater"""
-        subprocess.Popen([self.custom_updater], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        env = os.environ.copy()
+        custom_updater_args = self.custom_updater.split()
+        subprocess.Popen(custom_updater_args, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, env=env)
 
     def on_clicked_pm(self, *args) -> None:
         """start yumex"""
-- 
2.45.2

